include/zephyr/drivers/i2c.h

```diff
@@ -410,6 +410,35 @@ static inline bool i2c_is_ready_dt(const struct i2c_dt_spec *spec)
 	return device_is_ready(spec->bus);
 }
 
+/**
+ * @brief Dump out an I2C message
+ *
+ * Dumps out a list of I2C messages. For any that are writes (W), the data is
+ * displayed in hex. Setting dump_read will dump the data for read messages too,
+ * which only makes sense when called after the messages have been processed.
+ *
+ * It looks something like this (with name "testing"):
+ *
+ * @code
+ * D: I2C msg: testing, addr=56
+ * D:    W len=01: 06
+ * D:    W len=0e:
+ * D: contents:
+ * D: 00 01 02 03 04 05 06 07 |........
+ * D: 08 09 0a 0b 0c 0d       |......
+ * D:    W len=01: 0f
+ * D:    R len=01: 6c
+ * @endcode
+ *
+ * @param name Name of this dump, displayed at the top.
+ * @param msgs Array of messages to dump.
+ * @param num_msgs Number of messages to dump.
+ * @param addr Address of the I2C target device.
+ * @param dump_read Dump data from I2C reads, otherwise only writes have data dumped.
+ */
+void i2c_dump_msgs(const char *name, const struct i2c_msg *msgs,
```

Comment:
This is a stable API change and needs to go through the process.

Reply:
I looked at the process, clearly the work of a master bureaucrat, and nope.  That's never going to happen.

Since adding a parameter to one function is not possible, I guess adding another function is the way to do it.